novawallet/Common/URLHandling/Parsing/ClaimGiftUrlParsingService.swift

```diff
@@ -0,0 +1,175 @@
+            }
+
+            let publicKeyFetchRequest = GiftPublicKeyFetchRequest(
+                seed: seed,
+                ethereumBased: chain.isEthereumBased
+            )
+            let publicKey: Data = try self.giftPublicKeyProvider.getPublicKey(
+                request: publicKeyFetchRequest
+            )
+            let accountId = try chain.isEthereumBased
+                ? publicKey.ethereumAddressFromPublicKey()
+                : publicKey.publicKeyToAccountId()
+
+            return ClaimableGiftInfo(
+                seed: seed,
+                accountId: accountId,
+                chainAsset: chainAsset
+            )
+        }
+
+        mapOperation.addDependency(chainWrapper.targetOperation)
+
+        return chainWrapper.insertingTail(operation: mapOperation)
+    }
+
+    func createGiftClaimCheckWrapper(
+        dependingOn claimableGiftWrapper: CompoundOperationWrapper<ClaimableGiftInfo>
+    ) -> CompoundOperationWrapper<GiftClaimAvailabilityCheckResult> {
+        OperationCombiningService.compoundNonOptionalWrapper(operationQueue: operationQueue) {
+            let claimableGift = try claimableGiftWrapper.targetOperation.extractNoCancellableResultData()
+
+            return self.claimAvailabilityChecker.createAvailabilityWrapper(for: claimableGift)
+        }
+    }
+}
+
+// MARK: - OpenScreenUrlParsingServiceProtocol
+
+extension ClaimGiftUrlParsingService: OpenScreenUrlParsingServiceProtocol {
+    func parse(
+        url: URL,
+        completion: @escaping (Result<UrlHandlingScreen, OpenScreenUrlParsingError>) -> Void
+    ) {
+        guard
+            let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false),
+            let queryItems = urlComponents.queryItems
+        else {
+            completion(.failure(.openGiftClaimScreen(.invalidURL)))
+            return
+        }
+
+        let payloadString: String? = queryItems.first {
+            $0.name.trimmingCharacters(
+                in: .whitespacesAndNewlines
+            ) == UniversalLink.Gift.QueryKey.payload
+        }?.value
+
+        guard let payloadString else {
+            completion(.failure(.openGiftClaimScreen(.invalidURL)))
+            return
+        }
+
+        let wrapper = createWrapper(from: payloadString)
+
+        execute(
+            wrapper: wrapper,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: .main
+        ) { result in
+            switch result {
+            case let .success(navigation):
+                completion(.success(.giftClaim(navigation)))
+            case let .failure(error):
+                guard case let OpenScreenUrlParsingError.openGiftClaimScreen(serviceError) = error else {
+                    completion(.failure(.openGiftClaimScreen(.invalidURL)))
+                    return
+                }
+
+                completion(.failure(.openGiftClaimScreen(serviceError)))
+            }
+        }
+    }
+
+    func cancel() {}
```

Comment:
We probably want to store the wrapper from `parse` method in `CancellableCallStore` and cancel it here